src/utils/solidity/transactionFactory.ts

```diff
@@ -164,13 +225,49 @@ const transactionFactory: TransactionFactory = <Args, Result>(
     options = defaultOptions,
     environment = getGlobalEnvironment(),
   ) => {
-    const receipt = await sendTransaction(prepared, options, environment);
+    const receipt = await environment.eth
+      .sendTransaction(prepared.rawTransaction)
+      .then(null, error => {
+        throw new Error(
+          `Transaction failed for ${name}(${prepared.transactionArgs.join(
+            ', ',
+          )}): ${error.message}`,
+        );
+      });
+
+    const events = receipt.logs.reduce((carry, log) => {
```

Comment:
Manual transaction log parsing. This is needed because we loose the web3 transaction object that does this automatically for us through GraphQL and hardware wallets.

Anyways, our own solution is now able to parse all events that we have in our contracts, not only the one in the current ABI.

Reply:
Very cool!!